shared/src/ledger/masp.rs

```diff
@@ -200,3 +246,1044 @@ pub fn get_params_dir() -> PathBuf {
         masp_proofs::default_params_folder().unwrap()
     }
 }
+
+#[async_trait]
+pub trait ShieldedUtils : Sized + BorshDeserialize + BorshSerialize + Default + Clone {
+    async fn query_storage_value<T: Send>(
+        &self,
+        key: &storage::Key,
+    ) -> Option<T>
+    where
+        T: BorshDeserialize;
+
+    async fn query_epoch(&self) -> Epoch;
+
+    fn local_tx_prover(&self) -> LocalTxProver;
```

Comment:
Is this specific to the consumer? Something that would differ from what is being done on desktop/CLI?